Tests/my_test_Inference_RAxML.py

```diff
@@ -0,0 +1,264 @@
+from pathlib import Path
+import subprocess
+import pytest
+import random
+import os
+
+from Bio import Phylo
+from Bio.Phylo.Consensus import _BitString
+
+from mevolib.inference import _RAxML as Rax
+from mevolib._utils import NUMCORES
+
+
+class MockStdOut:
+    def __init__(self, tree_infile_path, tree_outfile_path, info_infile_path, info_outfile_path):
+        with open(tree_infile_path, "r") as fin:
+            self.stdout_tree = fin.read()
+        with open(tree_outfile_path, "w") as fin:
+            fin.write(self.stdout_tree)
+        with open(info_infile_path, "r") as fin:
+            self.stdout_info = fin.read()
+        with open(info_outfile_path, "w") as fin:
+            fin.write(self.stdout_info)
+
+    def get_mocked_tree_output(self):
+        return self.stdout_tree
+
+    def get_mocked_info_output(self):
+        return self.stdout_info
+
+
+class TestInferenceRAxML:
+    tmp_dir: Path = Path("tests/raxml_tmp_dir/").absolute()
+    manifest_dir: Path
+    # Couple of functions used to compare Phylo Trees:
+
+    # store and return all _BitStrings
+    def _bitstrs(self, tree):
+        bitstrs = set()
+        term_names = [term.name for term in tree.get_terminals()]
+        term_names.sort()
+        for clade in tree.get_nonterminals():
+            clade_term_names = [term.name for term in clade.get_terminals()]
+            boolvals = [name in clade_term_names for name in term_names]
+            bitstr = _BitString("".join(map(str, map(int, boolvals))))
+            bitstrs.add(bitstr)
+        return bitstrs
+
+    def compare(self, tree1, tree2):
+        term_names1 = [term.name for term in tree1.get_terminals()]
+        term_names2 = [term.name for term in tree2.get_terminals()]
+        # false if terminals or BitStrings are not the same
+        return set(term_names1) == set(term_names2) and self._bitstrs(tree1) == self._bitstrs(tree2)
```

Comment:
Do not replicate code! If you need this in all your tests, better add this as utils or pytest element.